winhlp32/hlpfile.c

```diff
@@ -2925,6 +2988,161 @@ static BOOL HLPFILE_SkipParagraph(HLPFILE *hlpfile, const BYTE *buf, const BYTE
+        {
+            // TODO: handle index
+            if (!strncmp(start, ":Title ", 7))
+            {
+                start += 7;
+                while (isspace(*start)) start++;
+                len = strlen(start);
+                cnt = HeapReAlloc(GetProcessHeap(), 0, cnt, sizeof(HLPFILE_PAGE) + (len + 1) * 2);
+                cnt->lpszTitle = ((BYTE *)cnt) + sizeof(HLPFILE_PAGE);
+                MultiByteToWideChar(hlpfile->codepage, 0, start, -1, cnt->lpszTitle, len);
+            }
+            str = next_str;
+            continue;
+        }
+        l = strtol(start, NULL, 10);
+        if ((l <= 0) || (l > 9))
+        {
+            str = next_str;
+            continue;
+        }
+        while (isdigit(*start)) start++;
+        while (isspace(*start)) start++;
+        end = strchr(start, '=');
+        if (!end)
+        {
+            if (l > curl) curl++;
+            else curl = l;
+        }
+        else if (l < curl) curl = l + 1;
+        if (curl == 1)
+            sprintf(tmp, "\\pard\\s%d ", curl);
+        else            
+            sprintf(tmp, "\\pard\\collapsed\\s%d ", curl);
+        if (!HLPFILE_RtfAddControl(&rd, tmp)) goto errexit;
+        if (end)
+        {
+            char *index = end + 1;
+            char *file = strchr(index, '@');
+            char *wnd = strchr(index, '>');
+            int w = -2;
+            *end = 0;
+            if (file)
+            {
+                *file = 0;
+                file++;
+            }
+            if (wnd)
+            {
+               *wnd = 0;
+               wnd++;
+               for (int w = hlpfile->numWindows - 1; w >= 0; w--)
+               {
+                   if (!stricmp(wnd, hlpfile->windows[w].name)) break;
+               }
+            }
+            HLPFILE_AllocLink(&rd, hlp_link_link, file ? file : hlpfile->lpszPath, -1, HLPFILE_Hash(index), FALSE, FALSE, w);
+            sprintf(tmp, "{\\field{\\*\\fldinst{ HYPERLINK \"%p\" }}{\\fldrslt{", rd.current_link);
+            if (!HLPFILE_RtfAddControl(&rd, tmp)) goto errexit;
+            rd.current_link = NULL;
+        }
+        else curl++;
+        // outline only works with utf8 codepage
+        MultiByteToWideChar(hlpfile->codepage, 0, start, -1, tmpW, 256);
+        tmpW[255] = 0;
+        WideCharToMultiByte(CP_UTF8, 0, tmpW, -1, tmp, 256, NULL, NULL);
+        tmp[255] = 0;
+        if (!HLPFILE_RtfAddControl(&rd, tmp)) goto errexit;
+        if (end && !HLPFILE_RtfAddControl(&rd, "}}}")) goto errexit;
+        if (!HLPFILE_RtfAddControl(&rd, "\\par")) goto errexit;
+        str = next_str;
+    }
+    if (!HLPFILE_RtfAddControl(&rd, "}")) goto errexit;
+    hlpfile->cnt_rtf = rd.data;
+    hlpfile->cnt_page = cnt;
+    cnt->file = hlpfile;
+    cnt->first_link = rd.first_link;
+    cnt->offset = rd.ptr - rd.data;
+    if (!cnt->lpszTitle) 
+    {
+        const WCHAR deftitle[] = {'C', 'o', 'n', 't', 'e', 'n', 't', 's', 0};
+        cnt->lpszTitle = deftitle;
```

Comment:
deftitle is on the stack.
```suggestion
        cnt->lpszTitle = (WCHAR*)L"Contents";
```